lib/developer_portal/app/controllers/developer_portal/admin/account/invoices_controller.rb

```diff
@@ -1,34 +1,88 @@
+# frozen_string_literal: true
+
 class DeveloperPortal::Admin::Account::InvoicesController < ::DeveloperPortal::BaseController
+  class PaymentGatewayNotSupported < StandardError; end
+
+  rescue_from PaymentGatewayNotSupported, with: :handle_not_found
 
   helper Finance::InvoicesHelper
   helper Accounts::InvoicesHelper
 
   before_action :authorize_finance
   before_action :find_provider
+  before_action :authorize_payment_gateway, only: %i[payment payment_succeeded]
+  before_action :find_invoice, only: %i[show payment payment_succeeded]
+
   activate_menu :account, :invoices
 
   liquify prefix: 'invoices'
 
   def index
-    collection = current_account.invoices.visible_for_buyer.page(params[:page])
-
+    collection = accessible_invoices.page(params[:page])
     invoices = Liquid::Drops::Invoice.wrap(collection)
     pagination = Liquid::Drops::Pagination.new(collection, self)
     assign_drops invoices: invoices, pagination: pagination
   end
 
   def show
-    invoice = current_account.invoices.visible_for_buyer.find(params[:id])
     assign_drops invoice: Liquid::Drops::Invoice.wrap(invoice)
   end
 
+  def payment
+    @payment_intent = @invoice.payment_intents.latest_pending.first!
+    @client_secret = stripe_payment_intent.client_secret
+    @stripe_publishable_key = payment_gateway_options[:publishable_key]
+  end
+
+  def payment_succeeded
+    @payment_intent = @invoice.payment_intents.find_by!(reference: stripe_payment_intent_params[:id])
+
+    service = Finance::StripePaymentIntentUpdateService.new(site_account, stripe_payment_intent)
+    if service.call
+      flash[:notice] = 'Payment transaction updated'
+    else
+      flash[:error] = 'Failed to update payment transaction'
```

Comment:
@damianpm Is this message shown to the user?

Reply:
Yes, it appears for a few seconds, and then it vanishes

Reply:
In this case, what can the user do to solve this error? 

If the user has alternatives to solve the error that triggers this message, I think the message should persist with a suggested solution. For example: _Failed to update payment transaction. Review your credit card details_

What do you think @damianpm ?

Reply:
@porueesq good remark! But flash messages should be short (technically hard to put many characters there). We should find another way of doing this, probably validations or something else.

Reply:
Yes @hallelujah , if the user needs to perform an action to correct the issue, we need another alternative to show the message.

Reply:
@damianpm Hence, the flash message is for the Admin Portal, and the screenshots are for the Developer Portal, right?

If these are correct, then no additional changes are necessary :+1: